libraries/botbuilder-dialogs-adaptive/src/recognizers/entityRecognizers/entityRecognizer.ts

```diff
@@ -6,14 +6,122 @@
  * Licensed under the MIT License.
  */
 
-import { Entity } from 'botbuilder-core';
-import { DialogContext } from 'botbuilder-dialogs';
+import { Activity, Entity, RecognizerResult } from 'botbuilder-core';
+import { DialogContext, Recognizer } from 'botbuilder-dialogs';
+import { TextEntity } from './textEntity';
 
-export interface EntityRecognizer {
-    recognizeEntities(
+/**
+ * Entity recognizers base class.
+ */
+export class EntityRecognizer extends Recognizer {
+    /**
+     * To recognize intents and entities in a users utterance.
+     *
+     * @param {DialogContext} dialogContext Dialog Context.
+     * @param {Partial<Activity>} activity Activity.
+     * @param {object} telemetryProperties Additional properties to be logged to telemetry with event.
+     * @param {object} telemetryMetrics Additional metrics to be logged to telemetry with event.
+     * @returns {Promise<RecognizerResult>} Analysis of utterance.
+     */
+    public async recognize(
+        dialogContext: DialogContext,
+        activity: Partial<Activity>,
+        telemetryProperties?: Record<string, string>,
+        telemetryMetrics?: Record<string, number>
+    ): Promise<RecognizerResult> {
+        // Identify matched intents
+        const text = activity.text || '';
+
+        const recognizerResult: RecognizerResult = {
+            text,
+            intents: {},
+        };
+
+        if (!text) {
+            // nothing to recognize, return empty result
+            return recognizerResult;
+        }
+
+        const textEntity = new TextEntity(text);
+        textEntity['start'] = 0;
```

Comment:
We should be using dot access whenever possible, can this be written as `textEntity.start = 0;`?

Reply:
start/end/score are not builtin properties in `TextEntity`, we use it by extending the `[key: string]: any` part of `Entity` interface. How can we avoid lint errors if writing these properties using dot access?

Reply:
<img width="464" alt="Screen Shot 2020-11-30 at 7 04 22 PM" src="https://user-images.githubusercontent.com/1235378/100692060-e18c7680-333e-11eb-9ef2-6ea8bb7aa9e0.png">

I don't see any lint errors in VS Code here - does `yarn lint` raise errors on this?

Reply:
Maybe it is `TextEntity` (class) not `Entity` (interface)?
![image](https://user-images.githubusercontent.com/946375/100692390-b6993580-33c5-11eb-96a5-4815de784c91.png)

Cast to `Entity` would resolve the error:
![image](https://user-images.githubusercontent.com/946375/100692491-eb0cf180-33c5-11eb-9505-e3d6faae62f9.png)

Reply:
As long as there is no issue in typing `textEntity` as `Entity` then I would do that.